crates/katana/rpc/src/utils/transaction.rs

```diff
@@ -89,6 +97,31 @@ pub fn compute_invoke_v1_transaction_hash(
     ])
 }
 
+pub fn compute_deploy_account_transaction_hash(
+    contract_address: FieldElement,
+    calldata: &[FieldElement],
+    class_hash: FieldElement,
+    contract_address_salt: FieldElement,
+    max_fee: FieldElement,
+    chain_id: FieldElement,
+    nonce: FieldElement,
+) -> FieldElement {
+    let mut inner_vec = calldata.to_vec(); 
+    inner_vec.push(class_hash);
+    inner_vec.push(contract_address_salt);
```

Comment:
its actually should be:

```rust
[class_hash, salt, constructor_calldata]
```

based on `starknet-rs` [impl](https://github.com/xJonathanLEI/starknet-rs/blob/35c287e1a06e6ab68447f5f0b9df53f910960f57/starknet-accounts/src/factory/mod.rs#L298)... which is different from the [starknet docs](https://docs.starknet.io/documentation/architecture_and_concepts/Blocks/transactions/#deployaccount_v1)